test/test_auth.py

```diff
@@ -16,9 +16,10 @@
 
 import os
 import sys
-import threading
 from urllib.parse import quote_plus
 
+from pymongo.io import threading
```

Comment:
We need to change all the test imports back to use stdlib threading. We can't ask all users to change what threading module they use since that would be a huge breaking change.

Reply:
We talked in the catch up meeting today about making our asyncio an explicit opt-in, similar to `gevent.monkey_patch()`.

Reply:
Even if we make it opt in, apps still need to be able to use the sync client from standard threads.

Reply:
Right, with the patch in place, `import threading` will be the patched version.

Reply:
I think that approach is a non-starter. We can't expect users to patch the stndlib just to use async pymongo.

Reply:
There isn't another way to ensure that the versions of threading, time, etc. that they import will work short of us telling them to import from us, similar to using [gevent](http://www.gevent.org/intro.html#monkey-patching).

If that is a non-starter then we're back to the drawing board.

Reply:
I don't understand. What happens when a stdlib thread runs a pymongo operation that internally uses greenletio? Are you saying that doesn't work?

Reply:
Correct, I verified that it blocks all other execution.

Reply:
Actually wait, what I actually did was use `time.time`.  Let me dig in again.

Reply:
Can you expand on "blocks all other execution"?

Reply:
I changed test_threading.py like this:
```
-from pymongo.io import threading
+import threading
```

And the tests pass and run in the same amount of time.

Reply:
If I change to stdlib import:

```python
Traceback (most recent call last):
  File "/Library/Frameworks/Python.Framework/Versions/3.10/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
starting sync!
waiting: 1691100697
    self.run()
  File "/Library/Frameworks/Python.Framework/Versions/3.10/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/steve.silvester/workspace/mongo-python-driver/green_test.py", line 38, in async_main_entry
    await_(async_main())
  File "/Users/steve.silvester/workspace/.venvs/mongo-python-driver/lib/python3.10/site-packages/greenletio/core.py", line 148, in await_
    parent = greenlet.getcurrent().parent or bridge.start()
  File "/Users/steve.silvester/workspace/.venvs/mongo-python-driver/lib/python3.10/site-packages/greenletio/core.py", line 39, in start
    if asyncio.get_event_loop().is_running():
  File "/Library/Frameworks/Python.Framework/Versions/3.10/lib/python3.10/asyncio/events.py", line 671, in get_event_loop
    raise RuntimeError('There is no current event loop in thread %r.'
RuntimeError: There is no current event loop in thread 'Thread-2 (async_main_entry)'.
/Library/Frameworks/Python.Framework/Versions/3.10/lib/python3.10/threading.py:1018: RuntimeWarning: coroutine 'test_database.<locals>.async_main' was never awaited
  self._invoke_excepthook(self)
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
find:    1691100697
Exception in thread Thread-1 (main):
Traceback (most recent call last):
  File "/Library/Frameworks/Python.Framework/Versions/3.10/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/Library/Frameworks/Python.Framework/Versions/3.10/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/steve.silvester/workspace/mongo-python-driver/green_test.py", line 66, in main
    time.sleep(1)
  File "/Users/steve.silvester/workspace/.venvs/mongo-python-driver/lib/python3.10/site-packages/greenletio/green/time.py", line 10, in sleep
    await_(asyncio.sleep(seconds))
  File "/Users/steve.silvester/workspace/.venvs/mongo-python-driver/lib/python3.10/site-packages/greenletio/core.py", line 152, in await_
    return parent.switch(coro_or_fn)
greenlet.error: cannot switch to a different thread
/Library/Frameworks/Python.Framework/Versions/3.10/lib/python3.10/threading.py:1018: RuntimeWarning: coroutine 'sleep' was never awaited
  self._invoke_excepthook(self)
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
```

We *might* be able to adapt `greenletio` to handle this case.

Reply:
I ran into the above error when I was using only `greenletio.green.socket` and stdlib for the rest, are you using all `greenletio` besides `thread`?

Reply:
Yeah, in your case you were using sync things in a thread, I verified that works.  It's when you want to do async things in a thread that things break down.  Which, you wouldn't really do anyway, so maybe we're okay?

Reply:
> Can you expand on "blocks all other execution"?

I was trying to mix `time.sleep(1)` with greenletio and it would not run the tasks concurrently.

Reply:
If you use sync pymongo in a thread while `greenletio` is present, things break?

Reply:
You can use sync pymongo in a thread with `greentletio` and stdlib threading, or async pymongo in a thread with `greenletio` threading.

Reply:
I believe those errors are expected. You can't call `await_` from a thread without setting up a loop just like you can't await an `async def` method outside of a running eventloop.

Reply:
The code appears to work fine with standard time+threading with this addition:
```python
    def async_main_entry():
        asyncio.set_event_loop(asyncio.new_event_loop())
        await_(async_main())
```

Reply:
Oh cool, yeah, that makes sense.